pkg/cel/cel.go

```diff
@@ -17,17 +17,12 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
-// CompileOrCELExpression attempts to compile cel expression strings into a cel.Program. If exprs contains more than one cel
-// expression string, it will surround each cell expression in parentheses and or them together.
-func CompileOrCELExpression(exprs ...string) (*cel.Program, error) {
-	exprs = FormatCelExprs(exprs...)
-	expr := strings.Join(exprs, " || ")
-	return CompileCELExpr(expr)
-}
+var mapStrDyn = types.NewMapType(types.StringType, types.DynType)
+var env *cel.Env
 
-func CompileCELExpr(expr string) (*cel.Program, error) {
-	mapStrDyn := types.NewMapType(types.StringType, types.DynType)
-	env, err := cel.NewEnv(
+func init() {
+	var err error
+	env, err = cel.NewEnv(
```

Comment:
I think having `mapStryDyn` and `env` as gobal vars is fine, but I don't think that we should initialize them in an `init` function because of the unrecoverable error.

instead I think you should use [sync.Once](https://pkg.go.dev/sync#Once).

you would create Once struct as a global var and then in `CompileCELExpr` you could have something like what is done in [newDatabase](https://github.com/kubearchive/kubearchive/blob/main/pkg/database/database.go#L48)

Reply:
> but I don't think that we should initialize them in an `init` function because of the unrecoverable error.

But it is an unrecoverable error, which is why there is a `panic`.  What is the point of returning an error and continuing?

Reply:
Yeah, as I look at documentation and comments for `init` vs. `sync.Once`, I disagree.

https://stackoverflow.com/questions/51953191/difference-between-init-and-sync-once-in-golang